lib/rexml/source.rb

```diff
@@ -68,14 +68,18 @@ def encoding=(enc)
     def read
     end
 
-    def match(pattern, cons=false)
+    def match(pattern, cons=false, read_source=false)
       if cons
         @scanner.scan(pattern).nil? ? nil : @scanner
       else
         @scanner.check(pattern).nil? ? nil : @scanner
       end
     end
 
+    def string=(string)
+      @scanner.string = string
+    end
```

Comment:
I'm OK with this for now but I want to remove this later. I want to just change scan pointer (`StringScanner#pos=`) instead of changing `@scanner.string`.

FYI: ruby/csv changes scan pointer: https://github.com/ruby/csv/blob/3ae9194409c426d4d9e5ea389959e29949e39c39/lib/csv/parser.rb#L56-L71

Reply:
OK!
I See.